sdk/trace/src/main/java/io/opentelemetry/sdk/trace/SdkSpanBuilder.java

```diff
@@ -191,7 +192,7 @@ public Span startSpan() {
     TraceState samplingResultTraceState =
         samplingResult.getUpdatedTraceState(parentSpanContext.getTraceState());
     SpanContext spanContext =
-        SpanContext.create(
+        ImmutableSpanContext.createBypassingValidation(
```

Comment:
By the way, any thoughts on storing a boolean based on `tracerSharedState.getIdGenerator().getClassName()` to only skip validation on our SDK ID generators?

Reply:
I was going to suggest something like this as well. I think that we probably should have this, to account for the case that you pointed out with a bad idGenerator letting things slip through unchecked.

It does make me think that there might be some value in providing a TCK for IdGenerator implementations, as well.